### PR TITLE
PS-8967 Replace keyring vault plugin with keyring vault component (8.1)

### PR DESCRIPTION
**Pull request by alina-derkach-oaza**
new file:   docs/use-keyring-file-vault-components.md
deleted:    docs/using-keyring-plugin.md
modified:   mkdocs-base.yml

### CODE REVIEW

**Review by patrickbirch — APPROVED** [review]
Good

File: docs/use-keyring-file-vault-components.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    {
+       "components": "file://component_keyring_vault"
+    }
+    ```
+
+    An example of `/lib/plugin/component_keyring_vault.cnf`:
+
+    ```json
+    {
+       "path": "/var/lib/c/keyring_vault", "read_only": false
```
Comment (oleksandr-kachan) [inline docs/use-keyring-file-vault-components.md]: "path" and "read_only" are not valid fields for component_keyring_vault config. You have valid ones listed below.

File: docs/use-keyring-file-vault-components.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    | Option       | Description            |
+    |--------------|------------------------|
+    | Command-line | --keyring-vault-config |
+    | Scope        | Global                 |
+    | Dynamic      | Yes                    |
+    | Data type    | Text                   |
+    | Default      |                        |
+
+    The `keyring_vault_config` file contains the following information:
```
Comment (oleksandr-kachan) [inline docs/use-keyring-file-vault-components.md]: There is no `keyring_vault_config` variable any more. Component config search is done similar to manifest file.
1) The component_keyring_vault config name is expected to be "component_keyring_vault.cnf".
2) There may be two configs - global one (located in 'plugin' directory), and instance specific (located in data directory).
3) Global config may have "read_local_config" option set which indicates server should search for instance specific config.

File: docs/use-keyring-file-vault-components.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    * `timeout` - the duration in seconds for the Vault server connection timeout. The default value is 15. The allowed range is from 0 to 86400. The timeout can be also disabled to wait an infinite amount of time by setting this variable to 0.
+
+    * `vault_url` - the Vault server address.
+
+    * `secret_mount_point` - the mount point name where the keyring_vault stores the keys.
+
+    * `token` - a token generated by the Vault server.
+
+    * `secret_mount_point_version [optional]` - the `KV Secrets Engine version (kv or kv-v2)` used.
```
Comment (oleksandr-kachan) [inline docs/use-keyring-file-vault-components.md]: Actual expected values are: "AUTO", "1", "2"

File: docs/use-keyring-file-vault-components.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+    * `vault_ca [optional]` - if the machine does not trust the Vault’s CA certificate, this variable points to the CA certificate used to sign the Vault’s certificates.
+
+    ??? example "Example of a configuration file"
+
+        ```text
+        timeout = 15
```
Comment (oleksandr-kachan) [inline docs/use-keyring-file-vault-components.md]: The config content must be a valid JSON

'''
{
    "timeout": 15,
    "vault_url": "https://vault.public.com:8202",
    "secret_mount_point": "secret",
    "secret_mount_point_version": "AUTO",
    "token": "58a20c08-8001-fd5f-5192-7498a48eaf20",
    "vault_ca": "/data/keyring_vault_confs/vault_ca.crt"
}
'''

**Comment by oleksandr-kachan** [comment]
Should we also mention component_keyring_kmip and component_keyring_kms in this doc? 
Just in case, upstream's keyring component installation doc - https://dev.mysql.com/doc/refman/8.0/en/keyring-component-installation.html.

**Review by oleksandr-kachan — CHANGES_REQUESTED** [review]
Please check comments above

File: docs/use-keyring-file-vault-components.md (hunk 1 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file-vault-components.md:39]: Should it be 

    ```json
    {
       "read_local_config": "/var/lib/c/keyring_vault"

File: docs/use-keyring-file-vault-components.md (hunk 3 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file-vault-components.md]: Added

File: docs/use-keyring-file-vault-components.md (hunk 4 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file-vault-components.md]: fixed

**Comment by alina-derkach-oaza** [comment]
> Should we also mention component_keyring_kmip and component_keyring_kms in this doc? Just in case, upstream's keyring component installation doc - https://dev.mysql.com/doc/refman/8.0/en/keyring-component-installation.html.

We have them in separate docs, i will add the links

File: docs/use-keyring-file-vault-components.md (hunk 1 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file-vault-components.md]: rewritten

File: docs/use-keyring-file-vault-components.md (hunk 2 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file-vault-components.md]: Rewritten

File: docs/use-keyring-file.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+!!! warning
+
+    The `keyring_file` plugin should not be used for regulatory compliance.
+
+To install the plugin, follow the [installing and uninstalling plugins] instructions.
+
+### Load the keyring plugin
+
+You should load the plugin at server startup with the `-early-plugin-load` option to enable keyrings.
```
Comment (oleksandr-kachan) [inline docs/use-keyring-file.md]: -early-plugin-load -> --early-plugin-load

File: docs/use-keyring-vault-component.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+```json
+{
+ "components": "file:///component_keyring_vault"
+}
+```
+
+The configuration settings are either in a global configuration file or a local configuration file.
+
+The `keyring_vault_config.cnf` file contains the following information:
```
Comment (oleksandr-kachan) [inline docs/use-keyring-vault-component.md]: keyring_vault_config.cnf -> component_keyring_vault.cnf

File: docs/use-keyring-vault-component.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+* `timeout` - the duration in seconds for the Vault server connection timeout. The default value is 15. The allowed range is from 0 to 86400. The timeout can be also disabled to wait an infinite amount of time by setting this variable to 0.
+
+* `vault_url` - the Vault server address.
+
+* `secret_mount_point` - the mount point name where the keyring_vault stores the keys.
+
+* `token` - a token generated by the Vault server.
+
+* `secret_mount_point_version [optional]` - the `KV Secrets Engine version (kv or kv-v2)` used. The allowed values are `AUTO`, `1`, and `2`.
```
Comment (oleksandr-kachan) [inline docs/use-keyring-vault-component.md]: Defaults to AUTO

Comment (alina-derkach-oaza) [reply docs/use-keyring-vault-component.md]: fixed

File: docs/use-keyring-vault-component.md (hunk 6 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-vault-component.md]: fixed

File: docs/use-keyring-file.md (hunk 5 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file.md]: fixed

**Review by oleksandr-kachan — APPROVED** [review]

File: docs/keyring-components-plugins-overview.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Keyring components and plugins overview
+
+Percona Server supports a keyring that enables internal server components and plugins to store sensitive information securely for later retrieval.
+
+!!! note
```
Comment (patrickbirch) [inline docs/keyring-components-plugins-overview.md]: should this be a warning?

File: docs/use-keyring-file.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+We recommend that you load the plugin in the configuration file to facilitate recovery for encrypted tables. Also, the redo log encryption and the undo log encryption cannot be used without `--early-plugin-load`. The normal plugin load happens too late at startup.
+
+To load the keyring_file plugin, run the following command:
+
+```{.bash data-prompt="$"}
+$ mysqld --early-plugin-load="keyring_file=keyring_file.so"
+```
+
+!!! note
```
Comment (patrickbirch) [inline docs/use-keyring-file.md]: This doesn't need to be a note - just text.

File: docs/use-keyring-vault-component.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
+# Use the keyring vault component
+
+The `keyring_vault` component extends the server capabilities and provides an interface for the database with a [HashiCorp Vault](https://www.hashicorp.com/products/vault/data-protection) server to store key and secure encryption keys. 
+
+!!! admonition "See also"
```
Comment (patrickbirch) [inline docs/use-keyring-vault-component.md]: Can we move this See also to the bottom?

File: docs/use-keyring-vault-component.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+      "timeout": 15,
+      "vault_url": "https://vault.public.com:8202",
+      "secret_mount_point": "secret",
+      "secret_mount_point_version": "AUTO",
+      "token": "58a20c08-8001-fd5f-5192-7498a48eaf20",
+      "vault_ca": "/data/keyring_vault_confs/vault_ca.crt"
+     }
+     ```
+
+!!! warning
```
Comment (patrickbirch) [inline docs/use-keyring-vault-component.md:69]: Can we move this to the secret mount point text?

File: snippets/keyring-components-installation.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+To install a keyring component, do the following:
+
+1. Write a manifest in a valid JSON format
+
+2. Write a configuration file
+
+A manifest file indicates which component to load. If the manifest file does not exist, the server does not load the component associated with that file. During startup, the server reads the global manifest file from the installation directory. The global manifest file can contain the required information or point to a local manifest file located in the data directory. If you have multiple server instances that use different keyring components use a local manifest file in each data directory to load the correct keyring component for that instance.
+
+!!! note
```
Comment (patrickbirch) [inline snippets/keyring-components-installation.md]: should be warning

File: docs/use-keyring-vault-component.md (hunk 11 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-vault-component.md:69]: The warning is about `secret_mount_point` and the text is about `secret_mount_point_version` parameter

File: snippets/keyring-components-installation.md (hunk 12 above)
Comment (alina-derkach-oaza) [reply snippets/keyring-components-installation.md]: fixed

File: docs/use-keyring-vault-component.md (hunk 10 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-vault-component.md]: moved

File: docs/use-keyring-file.md (hunk 9 above)
Comment (alina-derkach-oaza) [reply docs/use-keyring-file.md]: fixed

File: docs/keyring-components-plugins-overview.md (hunk 8 above)
Comment (alina-derkach-oaza) [reply docs/keyring-components-plugins-overview.md]: fixed

**Review by patrickbirch — APPROVED** [review]
good

**Pull request merged**